src/tmpfiles/tmpfiles.c

```diff
@@ -722,37 +746,49 @@ static int dir_cleanup(
-                                          format_timestamp_style(a, sizeof(a), atime_nsec / NSEC_PER_USEC, TIMESTAMP_US));
-                                continue;
+
+                        if (FLAGS_SET(age_by_file, AGE_BY_ATIME)) {
+                                if (atime_nsec != NSEC_INFINITY && atime_nsec >= cutoff_nsec) {
+                                        char a[FORMAT_TIMESTAMP_MAX];
+                                        log_debug("File \"%s\": access time %s is too new.",
+                                                  sub_path,
+                                                  format_timestamp_style(a, sizeof(a), atime_nsec / NSEC_PER_USEC, TIMESTAMP_US));
+
+                                        continue;
+                                }
                         }
 
-                        if (ctime_nsec != NSEC_INFINITY && ctime_nsec >= cutoff_nsec) {
-                                char a[FORMAT_TIMESTAMP_MAX];
-                                log_debug("File \"%s\": change time %s is too new.",
-                                          sub_path,
-                                          format_timestamp_style(a, sizeof(a), ctime_nsec / NSEC_PER_USEC, TIMESTAMP_US));
-                                continue;
+                        if (FLAGS_SET(age_by_file, AGE_BY_CTIME)) {
+                                if (ctime_nsec != NSEC_INFINITY && ctime_nsec >= cutoff_nsec) {
+                                        char a[FORMAT_TIMESTAMP_MAX];
+                                        log_debug("File \"%s\": change time %s is too new.",
+                                                  sub_path,
+                                                  format_timestamp_style(a, sizeof(a), ctime_nsec / NSEC_PER_USEC, TIMESTAMP_US));
+
+                                        continue;
+                                }
                         }
 
-                        if (btime_nsec != NSEC_INFINITY && btime_nsec >= cutoff_nsec) {
-                                char a[FORMAT_TIMESTAMP_MAX];
-                                log_debug("File \"%s\": birth time %s is too new.",
-                                          sub_path,
-                                          format_timestamp_style(a, sizeof(a), btime_nsec / NSEC_PER_USEC, TIMESTAMP_US));
-                                continue;
+                        if (FLAGS_SET(age_by_file, AGE_BY_BTIME)) {
+                                if (btime_nsec != NSEC_INFINITY && btime_nsec >= cutoff_nsec) {
+                                        char a[FORMAT_TIMESTAMP_MAX];
+                                        log_debug("File \"%s\": birth time %s is too new.",
+                                                  sub_path,
+                                                  format_timestamp_style(a, sizeof(a), btime_nsec / NSEC_PER_USEC, TIMESTAMP_US));
+
+                                        continue;
+                                }
```

Comment:
given that for dirs and files we really do the exact same 4 cecks now, i think we should split them out into a helper funciton and just call them from both places. That would make the code more featureful *and* shorter, and I love that.

Reply:
Good idea, will update. :)